src/RecoveryServices/RecoveryServices.SiteRecovery/NetworkInterfaceCard/NewAzureRmAsrVmNicIPConfig.cs

```diff
@@ -0,0 +1,171 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.RecoveryServices.SiteRecovery.Properties;
+using Microsoft.Azure.Management.RecoveryServices.SiteRecovery.Models;
+
+namespace Microsoft.Azure.Commands.RecoveryServices.SiteRecovery
+{
+    /// <summary>
+    ///     Creates Azure Site Recovery VM NIC IP configuration for A2A replication.
+    /// </summary>
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "RecoveryServicesAsrVMNicIPConfig", DefaultParameterSetName = ASRParameterSets.AzureToAzure, SupportsShouldProcess = true)]
+    [Alias("New-ASRVMNicIPConfig")]
+    [OutputType(typeof(IPConfigInputDetails))]
+    public class NewAzureRmAsrVmNicIPConfig : SiteRecoveryCmdletBase
+    {
+        #region Parameters
+        /// <summary>
+        ///    Gets or sets the IP config name.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
+            Mandatory = true,
+            HelpMessage = "Specify the IP config name.")]
+        public string IpConfigName { get; set; }
+
+        /// <summary>
+        ///     Gets or sets whether an existing IP config is selected for test failover/failover.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
+            Mandatory = false,
+            HelpMessage = "Specifies whether an existing IP config is selected for test failover/failover.")]
+        public SwitchParameter IsSelectedForFailover { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the name of the recovery subnet.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
+            Mandatory = false,
+            HelpMessage = "Specifies the name of the recovery subnet.")]
+        [ValidateNotNullOrEmpty]
+        public string RecoverySubnetName { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the static IP address that should be assigned to IP config on recovery.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
+            Mandatory = false,
+            HelpMessage = "Specifies the IP address of the recovery IP config.")]
+        [ValidateNotNull]
+        public string RecoveryStaticIPAddress { get; set; }
+
+        /// <summary>
+        ///     Gets or sets the id of the public IP address resource associated with the recovery IP config.
+        /// </summary>
+        [Parameter(ParameterSetName = ASRParameterSets.AzureToAzure,
+            Mandatory = false,
+            HelpMessage = "Specifies the ID of the public IP address associated with the recovery IP config.")]
+        [ValidateNotNullOrEmpty]
```

Comment:
as mentioned update this and tfo public ip

Reply:
done